src/Castle.Core.Tests/DirectoryUtil.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2004-2010 Castle Project - http://www.castleproject.org/
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle
+{
+	using System;
+	using System.IO;
+	using System.Reflection;
+	using System.Reflection.Emit;
+
+	using NUnit.Framework;
+
+	public class DirectoryUtil
+	{
+		public static string GetCurrentDirectory(string relativePath = null)
+		{
+			// GVDM: HACK: This is not complete, it will trash dotnet core tests in TeamCity, once I finally get tests netcoreapp1.0 compliant, I consider this to be a hack to keep me going. The long term goal for this class is to combat pathing changes for tests at a framework level. Tests should always use DirectoryUtil.GetCurrentDirectory for 'base' pathing. 
+
+			var baseDirectory = AppDomain.CurrentDomain.BaseDirectory;
```

Comment:
NUnit 3.0's [breaking changes](https://github.com/nunit/docs/wiki/Breaking-Changes) doc lists the following, does `TestDirectory` not do what you need:

> CurrentDirectory: No longer set to the directory containing the test assembly. Use `TestContext.CurrentContext.TestDirectory` to locate that directory.

What about using Rob's suggestion (https://github.com/nunit/nunit/issues/1072#issuecomment-218725978) of setting the current directory in a static constructor of a test fixture, that would basically reimplement what NUnit 2.x was doing for us without needing to change tests all over the place?